src/emulator/controller.ts

```diff
@@ -327,19 +327,23 @@ export async function startAll(options: any, showUI: boolean = true): Promise<vo
   const targets = filterEmulatorTargets(options);
   options.targets = targets;
 
-  const projectId: string | undefined = getProjectId(options, true);
-
   if (targets.length === 0) {
     throw new FirebaseError(
       `No emulators to start, run ${clc.bold("firebase init emulators")} to get started.`
     );
   }
+  const hubLogger = EmulatorLogger.forEmulator(Emulators.HUB);
+  hubLogger.logLabeled("BULLET", "emulators", `Starting emulators: ${targets.join(", ")}`);
+
+  const projectId: string | undefined = getProjectId(options, true);
+  if (Constants.isFakeProject(projectId)) {
+    hubLogger.logLabeled(
+      "BULLET",
+      "emulators",
+      `Detected fake project ID "${projectId}", emulated services will use a demo configuration and non-emulated services will fail.`
```

Comment:
Did we actually block off all access to non-emulated services?

Reply:
No but since we are sure the project can't exist we're also sure it can't be accessed. I'll clarify.